### PR TITLE
Add standard JSO-compliant README

### LINKED ISSUE #2: Move README info to the doc?
It feels more in the spirit of the package to have a generic README that would a JSO-compliant solver typically have, and move all the extra information to the documentation of this package.

### PR DESCRIPTION
**Pull request by tmigot**
Close #2 
The removed text is moved to SolverCore.jl https://github.com/JuliaSmoothOptimizers/SolverCore.jl/pull/81

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,43 +13,58 @@
...
+It uses other [JuliaSmoothOptimizers](https://jso.dev) packages for development.
+In particular, [NLPModels.jl](https://github.com/JuliaSmoothOptimizers/NLPModels.jl) is used for defining the problem, and [SolverCore](https://github.com/JuliaSmoothOptimizers/SolverCore.jl) for the output.
+
+## How to Cite
+
+If you use JSOSolverSkeleton.jl in your work, please cite using the format given in [CITATION.cff](https://github.com/JuliaSmoothOptimizers/JSOSolverSkeleton.jl/blob/main/CITATION.cff).
+
+## Installation
+
+<p>
```
Comment (abelsiqueira) [inline README.md:38]: Why do you need html here?

Comment (tmigot) [reply README.md:38]: Just to have the nice Julia small logo :)

Comment (abelsiqueira) [reply README.md:38]: Ah because you want to set the width, right? Otherwise just normal markdown should be fine

**Review by abelsiqueira — APPROVED** [review]
Thanks

File: README.md (hunk 1 above)
Comment (tmigot) [reply README.md:38]: Yes, this is for the width.

**Pull request merged**